pkg/controller/rds/dbcluster/setup.go

```diff
@@ -0,0 +1,93 @@
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/rds/v1alpha1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupDBCluster adds a controller that reconciles DbCluster.
+func SetupDBCluster(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.DBClusterGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.preCreate = preCreate
+			e.preDelete = preDelete
+			e.filterList = filterList
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.DBCluster{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.DBClusterGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.DescribeDBClustersInput) error {
+	obj.DBClusterIdentifier = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+// This probably requires custom Conditions to be defined for handling all statuses
+// described here https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Status.html
+// Need to get help from community on how to deal with this. Ideally the status should reflect
+// the true status value as described by the provider.
+func postObserve(_ context.Context, cr *svcapitypes.DBCluster, resp *svcsdk.DescribeDBClustersOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	switch aws.StringValue(resp.DBClusters[0].Status) {
+	case "available":
+		cr.SetConditions(xpv1.Available())
+	case "deleting", "stopped", "stopping":
+		cr.SetConditions(xpv1.Unavailable())
+	case "creating":
+		cr.SetConditions(xpv1.Creating())
+	}
+	return obs, nil
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.CreateDBClusterInput) error {
+	obj.DBClusterIdentifier = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.DeleteDBClusterInput) (bool, error) {
+	obj.DBClusterIdentifier = aws.String(meta.GetExternalName(cr))
+	return false, nil
+	// TODO, guessing that the bool returned here should check the orphan field
+}
+
+func filterList(cr *svcapitypes.DBCluster, obj *svcsdk.DescribeDBClustersOutput) *svcsdk.DescribeDBClustersOutput {
+	clusterIdentifier := aws.String(meta.GetExternalName(cr))
+	resp := svcsdk.DescribeDBClustersOutput{}
+	for _, dbCluster := range obj.DBClusters {
+		if dbCluster.DBClusterIdentifier == clusterIdentifier {
```

Comment:
```suggestion
		if aws.StringValue(dbCluster.DBClusterIdentifier) == aws.StringValue(clusterIdentifier) {
```
Both of these are pointers, meaning equality check is done on their memory address. However, we should check if their values match. That's why it always says the resource is not found, hence try to create it over and over.